File: javascript/searching/linearSearch.js
/**
 * Linear Search
 * Time:  O(n)
 * Space: O(1)
 */

function linearSearch(arr, target) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === target) return i;
  }
  return -1;
}

module.exports = linearSearch;

File: python/searching/linear_search.py
"""
Linear Search
--------------
Time:  O(n)
Space: O(1)

Just walks through the array one element at a time.
Not efficient but works on unsorted data, which is its main advantage
over binary search.
"""


def linear_search(arr, target):
    for i, val in enumerate(arr):
        if val == target:
            return i
    return -1
